### PR TITLE
Display file attribute values in dwarfdump

### PR DESCRIPTION
**Pull request by philipc**
Also some other small fixes.

### CODE REVIEW

File: examples/dwarfdump.rs [hunk 1]
```diff
@@ -134,24 +134,26 @@ fn dump_info<Endian>(file: &object::File,
...
+            dump_entries(offset,
+                         unit.entries(&abbrevs),
                          unit.address_size(),
                          unit.format(),
                          debug_line,
                          debug_loc,
                          debug_ranges,
                          debug_str,
                          flags);
+            offset += unit.length_including_self();
```
Comment (fitzgen) [inline examples/dwarfdump.rs]: I think the offset stuff should be moved within the unit eventually, but doesn't need to be now.

File: examples/addr2line.rs [hunk 2]
```diff
@@ -165,10 +165,9 @@ impl Unit {
...
 
-        let comp_dir = match entry.attr_value(gimli::DW_AT_comp_dir) {
-                Some(gimli::AttributeValue::String(dir)) => Some(dir),
-                Some(gimli::AttributeValue::DebugStrRef(offset)) => debug_str.get_str(offset).ok(),
-                _ => None,
+        let comp_dir = match entry.attr(gimli::DW_AT_comp_dir) {
+                Some(attr) => attr.string_value(debug_str),
+                None => None,
             }
             .map(|dir| dir.to_string_lossy().into_owned());
```
Comment (philipc) [inline examples/addr2line.rs]: Thanks, I'm still not familiar with these chaining functions.

File: src/unit.rs [hunk 3]
```diff
@@ -1529,6 +1529,17 @@ impl<'input, Endian> Attribute<'input, Endian>
             _ => return None,
         })
     }
+
+    /// Try to convert this attribute's value to a string.
```
Comment (philipc) [inline src/unit.rs]: I slightly reworded the short description, and used your text as the body. I wanted to keep the short description similar to the other functions.

File: examples/dwarfdump.rs (hunk 1 above)
Comment (philipc) [inline examples/dwarfdump.rs]: Ok. Doing it here is a bit ugly, but I wasn't sure if the offset stuff was going to be useful outside of this dwarfdump clone.

**Pull request merged**